src/commands/up.rs

```diff
@@ -127,10 +135,25 @@ impl UpCommand {
                     .as_deref()
                     .unwrap_or_else(|| DEFAULT_MANIFEST_FILE.as_ref());
                 let bindle_connection = self.bindle_connection();
-                spin_loader::from_file(manifest_file, &working_dir, &bindle_connection).await?
+                spin_loader::from_file(
+                    manifest_file,
+                    if self.direct_mounts {
+                        None
```

Comment:
Just a random thought about further refactoring - does it make sense to introduce `DirectMounts` for `WorkingDirectory` enum which `.path()` resolves to `None`?

https://github.com/fermyon/spin/blob/8dfc4d6fbb31d624eb034f78640cb207bec1f682/src/commands/up.rs#L221-L233

(Probably, `WorkingDirectory` could be provided instead of `base_dst`, but I haven't checked for does it fit or not.)